cmd/main_test.go

```diff
@@ -301,3 +307,105 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("TestGetWorkspacesWithAccess", func(gv string, allNamespaces []k8sapi.Namespace, expectedWorkspaces []crt.Workspace) {
+	e := echo.New()
+	c := e.NewContext(nil, nil)
+	Context("When namespaces ns1 and ns2 has all the necessary permissions", func() {
+		BeforeEach(func() {
+			gv = "v1alpha1"
+			namespaceNames := []string{"test-tenant", "test-tenant-2"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+			}			
+			createRole(k8sClient, "test-tenant", "namespace-access", []string{"create", "list", "watch", "delete"})
+			createRole(k8sClient, "test-tenant-2", "namespace-access-2", []string{"create", "list", "watch", "delete"})
+			createRoleBinding(k8sClient, "namespace-access-user-binding", "test-tenant", "user1@konflux.dev", "namespace-access")
+			createRoleBinding(k8sClient, "namespace-access-user-binding-2", "test-tenant", "user2@konflux.dev", "namespace-access")
+			expectedWorkspaces = []crt.Workspace{
+				{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Workspace",
+						APIVersion: gv,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-tenant",
+					},
+					Status: crt.WorkspaceStatus{
+						Namespaces: []crt.SpaceNamespace{
+							{
+								Name: "test-tenant",
+								Type: "default",
+							},
+						},
+					},
+				},
+				{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Workspace",
+						APIVersion: gv,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-tenant-2",
+					},
+					Status: crt.WorkspaceStatus{
+						Namespaces: []crt.SpaceNamespace{
+							{
+								Name: "test-tenant-2",
+								Type: "default",
+							},
+						},
+					},
+				},
+			}
+		})
+
+		It("WorkspaceList should be created with ns1 and ns2 names", func() {
```

Comment:
Let's try rephrasing it a tiny bit to make the schema more readable
something similar to this:
  Describe("GetWorkspacesWithAccess querying for workspaces with access"  ) {
Context("When workspace "test-tenant"'s namespaces has  all the necessary permissions",) { 
It("Should return a WorkspaceList with test-tenant workspace and ns1 and ns2 in it")

Reply:
(same for the rest of the tests)